### PR TITLE
chore: no-root container as well as the update of the alpine image

### PR DESCRIPTION
**Pull request by qdupuy**
Update of the alpine image in order to have the latest functionalities of the tools of each database (example: postgresql-client)

Setting up the no-root in the container

To reproduce my work :

Clone the repo while being in my branch

Do the commands :

```
docker-compose build --no-cache
docker-compose up minio mariadb postgres mongodb -d
```

Connect to the minio to create the "backup" bucket

Then, we can launch the backup of each database :

```
docker-compose up mariadb-backup mongodb-backup postgres-backup
```

The tasks will end in success :

```
job-backup-db-postgres-backup-1 exited with code 0
job-backup-db-mariadb-backup-1 exited with code 0
job-backup-db-mongodb-backup-1 exited with code 0
```

If you want to see the current user, add the `whoami` command to the `cli_help` function before exit 1

If you modify the main.sh file, don't forget to rebuild the image

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -79,7 +79,7 @@ services:
       BAGCLI_DATABASE_PORT: 3306
       
   postgres:
-    image: postgres:11
+    image: postgres:latest
```
Comment (agravelot) [inline docker-compose.yml:82]: Why not using a matrix to test with multiple Postgres versions ? Probably the same for other databases.

File: Dockerfile [hunk 2]
```diff
@@ -13,13 +13,20 @@ WORKDIR /backup-cli
...
 COPY src/ ./
 
-RUN chmod +x /usr/bin/backup \
-  && chmod +x -R ./commands
+RUN chmod +x /usr/bin/backup; \
+    chmod +x -R ./commands; \
+    chown -R job /usr/bin/backup; \
+    chown -R job ./commands
+
+USER job
```
Comment (liogate) [inline Dockerfile:29]: Pourquoi ne pas mettre le USER avant le COPY ?

File: Dockerfile [hunk 3]
```diff
@@ -13,13 +13,20 @@ WORKDIR /backup-cli
...
 
 COPY main.sh /usr/bin/backup
 COPY src/ ./
 
-RUN chmod +x /usr/bin/backup \
-  && chmod +x -R ./commands
+RUN chmod +x /usr/bin/backup; \
+    chmod +x -R ./commands; \
+    chown -R job /usr/bin/backup; \
+    chown -R job ./commands
```
Comment (liogate) [inline Dockerfile]:
```suggestion
RUN chmod +x /usr/bin/backup \
    && chmod +x -R ./commands \
    && chown -R job /usr/bin/backup \
    && chown -R job ./commands
```

**Review by liogate — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (qdupuy) [reply Dockerfile:29]: L'utilisateur job n'a pas les droits nécessaires pour le chown/chmod ni dans le répertoire /usr/bin/

```
#17 0.126 chmod: /usr/bin/backup: Operation not permitted
```

**Pull request merged**